homeassistant/components/camera/__init__.py

```diff
@@ -777,6 +784,47 @@ async def handle(self, request: web.Request, camera: Camera) -> web.StreamRespon
             raise web.HTTPBadRequest() from err
 
 
+class Annotation:
+    """Text to be added to snapshot image."""
+
+    annotation_padding: int = 10
+
+    def __init__(self, service_call: ServiceCall, hass: HomeAssistant) -> None:
+        """Initialize annotation with ServiceCall and Hass."""
+        if "annotation" in service_call.data:
+            self.text = service_call.data["annotation"]
+            self.hass = hass
+            self.is_defined = True
+        else:
+            self.is_defined = False
+
+    def annotate(self, image_stream:bytes) -> bytes:
+        """Write annotation to the file provided as byte stream direct from camera feed."""
+        self.text.hass = self.hass
+        annotation_text = self.text.async_render()
+        _LOGGER.debug(
+            "Annotating snapsot with template text: %s",
+            annotation_text,
+        )
+        image_buffer = io.BytesIO(image_stream) 
+        img = PilImage.open(image_buffer)
```

Comment:
Looks good. Can this deal with all image types e.g. SVG?  I have a series of samples in the generic camera tests directory, maybe you could verify they work in some pytests?

Reply:
I cant get to write anything but corrupt data this way, so I'll have to put this on pause for bit as Ive exhausted my time for now.

Reply:
I am not sure, but to be honest I also have not seen any existing test cases for svg nor aware of that as a use case for Camera component.

Reply:
If you do PilImage.open() on an SVG file or byte stream, I think you will get an error.  So you either need to explicitly block this with some sort of not supported exception/error message or find a way to support it.